feat(dash): Enable Stateful Mode and rsq Corruptions for DASH

Kajlid opened the pull request:
- Imported STATEFUL and newState utilities for DASH (segment.ts, dashManifestUtils.ts and dashManifestUtils.test.ts) to support stateful mode.
- Ensured rsq corruptions are possible for DASH when the proxy is running in stateful mode, similar to how rsq corruptions are enabled for HLS.

This should solve the issue #63: Support for relative corruptions (rsq) for DASH live streams.

Nfrederiksen reviewed (changes requested):
It looks good. The feature is indeed implemented and behaves as expected 👍 
Only a few thing I'd need before approving.

1. `.history` can be included in `.gitignore` as well. No need to include it in the main branch.
2. Now sure in what update it got missing but add back `import { URLSearchParams } from 'url';` in utils.ts, 
just to be consistent in the project as it is now currently using the global `URLSearchParams` in that script when without the import. 
3. I like the handling of the case when a relative sequence corruption is attempted but the proxy is not in a stateful mode in `config.ts`.
 And although it is originally outside of the scope of the original issue #63 , it would be much appreciated if
a unit test could be added for that case in `config.test.ts` method `getAllManifestConfigs`. 
It needs to test if isDash AND !STATEFUL, and also case if !isDash AND !STATEFUL. This would be super nice to have in place :) 
4. Lastly, after all other points are completed. One final task is to update the READme file with the new feature and an example url. Just follow other examples. :)

Kajlid commented:
Hi Nicholas!
Thank you for your answer and for giving feedback in such a structured and understandable way :). I have updated the code with the changes you suggested, including the new unit test and updated README. 

I did not quite understand the first sentence of point 3, since the code I added in config.ts was inspired from the similar if statement above, giving an error message if !STATEFUL and !isDash, so I added a similar error message for !STATEFUL and isDash. This is also what the new unit test in config.test.ts is checking for, those error messages. In other words: I added code in config.test.ts now, but did not understand if I also need to make some change in config.ts.

Thanks again for the feedback and please get back if I need to make additional changes.

Nfrederiksen reviewed (approved):
This looks good to me.
